internal/runtimehandlerhooks/utils.go

```diff
@@ -121,3 +125,74 @@ func UpdateIRQSmpAffinityMask(cpus, current string, set bool) (cpuMask, bannedCP
 	}
 	return maskStringWithComma, invertedMaskStringWithComma, nil
 }
+
+func restartIrqBalanceService() error {
+	return exec.Command("systemctl", "restart", "irqbalance").Run()
+}
+
+func updateIrqBalanceConfigFile(irqBalanceConfigFile, newIRQBalanceSetting string) error {
+	input, err := ioutil.ReadFile(irqBalanceConfigFile)
+	if err != nil {
+		return err
+	}
+	lines := strings.Split(string(input), "\n")
+	found := false
+	for i, line := range lines {
+		if strings.HasPrefix(line, irqBalanceBannedCpus+"=") {
+			lines[i] = irqBalanceBannedCpus + "=" + "\"" + newIRQBalanceSetting + "\"" + "\n"
+			found = true
+		}
+	}
+	output := strings.Join(lines, "\n")
+	if !found {
+		output = output + "\n" + irqBalanceBannedCpus + "=" + "\"" + newIRQBalanceSetting + "\"" + "\n"
+	}
+	if err := ioutil.WriteFile(irqBalanceConfigFile, []byte(output), 0644); err != nil {
+		return err
+	}
+	return nil
+}
+
+func retrieveIrqBannedCPUMasks(irqBalanceConfigFile string) (string, error) {
+	input, err := ioutil.ReadFile(irqBalanceConfigFile)
+	if err != nil {
+		return "", err
+	}
+	lines := strings.Split(string(input), "\n")
+	for _, line := range lines {
+		if strings.HasPrefix(line, irqBalanceBannedCpus+"=") {
+			return strings.Trim(strings.Split(line, "=")[1], "\""), nil
+		}
+	}
+	return "", nil
+}
+
+func fileExists(filename string) bool {
+	info, err := os.Stat(filename)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return !info.IsDir()
+}
+
+func isAllMaskSet(in []byte) bool {
```

Comment:
Can you please add the comment on the method, that will describe the purpose of the method?

Reply:
of course, done.